### PR TITLE
Feature/guardian connect 2

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes #<number>

## Submitter Checklist:

- [ ] *Unit Tests* are updated to cover new or changed functionality
- [ ] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/BraveVPN/BraveVPN.swift [hunk 1]
```diff
@@ -33,9 +40,13 @@ class BraveVPN {
       if let error = error {
         logAndStoreError("Failed to load vpn conection: \(error)")
       }
-
-      GRDGatewayAPI.shared()._loadCredentialsFromKeychain()
-
+      
+      GRDSubscriptionManager.setIsPayingUser(true)
```
Comment (tzeejay) [inline Client/Frontend/BraveVPN/BraveVPN.swift:44]: Is this intentional? You should only mark users as being paying users if the receipt includes a valid subscription line item

File: Client/Frontend/BraveVPN/BraveVPN.swift [hunk 2]
```diff
@@ -33,9 +40,13 @@ class BraveVPN {
...
+      helper.verifyMainCredentials { _, _ in }
```
Comment (tzeejay) [inline Client/Frontend/BraveVPN/BraveVPN.swift:45]: I would recommend to log the return value here

File: Client/Frontend/BraveVPN/BraveVPN.swift [hunk 3]
```diff
@@ -557,11 +290,91 @@ class BraveVPN {
 
   static func clearCredentials() {
     GRDKeychain.removeGuardianKeychainItems()
-    GRDKeychain.removeKeychanItem(forAccount: kKeychainStr_SubscriberCredential)
+    GRDKeychain.removeSubscriberCredential(withRetries: 3)
```
Comment (tzeejay) [inline Client/Frontend/BraveVPN/BraveVPN.swift:293]: When is this called? When doing this are you also trying to remove the VPN credentials?

**Comment by iccub** [comment]
due to big architecture overhauld of the codebase work will continue on this branch https://github.com/brave/brave-ios/pull/5495

**Pull request closed without merging**